dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp

```diff
@@ -1637,19 +1640,194 @@ TEST(DataInterfaceIIDMTest, testImportError) {
 }
 
 TEST(DataInterfaceIIDMTest, testImportExport) {
-  const powsybl::iidm::Network& network = createNodeBreakerNetworkIIDM();
+  auto network = createNodeBreakerNetworkIIDM();
+
+  shared_ptr<DataInterfaceIIDM> dataOutput = createDataItfFromNetwork(createNodeBreakerNetworkIIDM());
+  ASSERT_NO_THROW(dataOutput->dumpToFile("network.xml"));
+  const powsybl::iidm::Network& outputNetwork = dataOutput->getNetworkIIDM();
+  ASSERT_THROW_DYNAWO(dataOutput->dumpToFile(".."), Error::GENERAL, KeyError_t::XmlFileParsingError);
+
+  shared_ptr<DataInterface> dataInput = DataInterfaceIIDM::build("network.xml");
+  shared_ptr<DataInterfaceIIDM> dataInputIIDM = boost::dynamic_pointer_cast<DataInterfaceIIDM>(dataInput);
+  const powsybl::iidm::Network& inputNetwork = dataInputIIDM->getNetworkIIDM();
+
+  ASSERT_EQ(outputNetwork.getId(), inputNetwork.getId());
+  ASSERT_EQ(outputNetwork.getId(), network->getId());
+  ASSERT_EQ(inputNetwork.getId(), network->getId());
+}
+
+TEST(DataInterfaceIIDMTest, testClone) {
+  auto network = createNodeBreakerNetworkIIDM();
+
+  shared_ptr<DataInterfaceIIDM> data = createDataItfFromNetwork(network);
+  shared_ptr<DataInterfaceIIDM> data2 = boost::dynamic_pointer_cast<DataInterfaceIIDM>(data->clone());
```

Comment:
only test copy constructor and not assignment

Reply:
Aim is to test the clone, not the operators. Should I put the operators in private then ?

Reply:
any method that exists need to be tester properly

Reply:
If they are private they cannot be tested

Reply:
either they are used and needs to be tested either they are useless and should be removed no?